### PR TITLE
Servo gimbal

### PR DESCRIPTION
**Pull request by anton-matosov**
Added servo gimbal class provided by @LorenzMeier 
Enabled attitude compensation by default
Implemented yaw compensation
Added missing MAVLink enum definitions for Gimbal related operations
Got rid of float equality checks in Gimbal class

Defined gimbal mixer scales for the "default" SK450 DeadCat inline servo gimbal with recommended servos. 

Made aux output pwm rate, min, max, disarmed and failsafe values configurable the same way as for the main outputs

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Its a bit surprising most CI systems failed on this PR - is there some build breakage?

**Comment by LorenzMeier** [comment]
Ah yes - you need to rebase on master. The CI tools don't accept ancient branches.

**Comment by anton-matosov** [comment]
This PR was rebased to the latest master right before creating it. This is the branch parent commit:
Commit: cf7f59e1f32977e49be9367b6441076ef77fa884 [cf7f59e]
Parents: 0261f5dfbe, dfd6be78bf
Author: Lorenz Meier <lorenz@px4.io>
Date: March 3, 2015 at 9:33:11 AM PST
Labels: origin/master origin/HEAD master

Merge pull request #1874 from DonLakeFlyer/RCParams
Add RC_CHAN_CNT, RC_TH_USER for QGC

**Comment by anton-matosov** [comment]
> git -c core.askpass=true fetch --tags --progress https://github.com/PX4/Firmware +refs/pull/*:refs/remotes/origin/pr/*
ERROR: Timeout after 10 minutes
ERROR: Error fetching remote repo 'origin'
hudson.plugins.git.GitException: Failed to fetch from https://github.com/PX4/Firmware

**Comment by anton-matosov** [comment]
Cloning into 'Tools/genmsg'...

No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.

The build has been terminated

**Comment by anton-matosov** [comment]
It loos like there were some problems with the git repos access.

**Comment by anton-matosov** [comment]
@LorenzMeier all build issues are gone.

File: ROMFS/px4fmu_common/init.d/rc.interface [hunk 1]
```diff
@@ -119,10 +120,43 @@ then
 	then
 		if fmu mode_pwm
 		then
-			mixer load /dev/pwm_output1 $MIXER_AUX_FILE
+			mixer load $OUTPUT_AUX_DEV $MIXER_AUX_FILE
 		else
 			tone_alarm $TUNE_ERR
 		fi
+
+		if [ $PWM_AUX_OUT != none ]
```
Comment (LorenzMeier) [inline ROMFS/px4fmu_common/init.d/rc.interface]: Is the NuttX default of an unset variable none? Else please add ```set PWM_AUX_OUT none``` in the top level rcS file where PWM_OUT is set.

File: src/drivers/gimbal/gimbal.cpp [hunk 2]
```diff
@@ -0,0 +1,542 @@
...
+			pitch += 1.0f / M_PI_F * M_DEG_TO_RAD_F * cmd.param2;
+			yaw += 1.0f / M_PI_F * M_DEG_TO_RAD_F * cmd.param3;
+			
+			updated = true;
+
+		}
+
+#if 0
+		// XXX change this to the real quaternion command
+		if (cmd.command == VEHICLE_CMD_DO_MOUNT_CONTROL_QUAT && mountMode == VEHICLE_MOUNT_MODE_MAVLINK_TARGETING) {
```
Comment (LorenzMeier) [inline src/drivers/gimbal/gimbal.cpp]: Would you be interested in implementing this as well? The math library has quaternion conversion functions here:
https://github.com/PX4/Firmware/blob/master/src/lib/mathlib/math/Quaternion.hpp#L204
and from DCM to Euler:
https://github.com/PX4/Firmware/blob/master/src/lib/mathlib/math/Matrix.hpp#L475

Comment (LorenzMeier) [inline src/drivers/gimbal/gimbal.cpp]: The reason I'm suggesting this is that gimbal lock (no pun intended) is a real issue depending on the mounting location of the Gimbal, and we're trying to do gimbal math in Quaternion space only on the interface level.

Comment (anton-matosov) [inline src/drivers/gimbal/gimbal.cpp]: Ok, will do this in couple of days

File: ROMFS/px4fmu_common/init.d/rc.interface (hunk 1 above)
Comment (anton-matosov) [inline ROMFS/px4fmu_common/init.d/rc.interface]: Added them in the same commit https://github.com/anton-matosov/Firmware/commit/2b13cd3706fb1ffe569c1dbc3e7a42425b4be6bc

File: src/drivers/gimbal/gimbal.cpp (hunk 2 above)
Comment (anton-matosov) [inline src/drivers/gimbal/gimbal.cpp]: @LorenzMeier, can you point me to some resources where these MAVLink messages are discussed/documented, please? I would like to understand them better before I get them implemented.
Another question is how to debug this messages?

Comment (dagar) [inline src/drivers/gimbal/gimbal.cpp]: @anton-matosov Would this help? http://mavlink.org/messages/common

Comment (anton-matosov) [inline src/drivers/gimbal/gimbal.cpp]: These docs are generated from the same xml files as used for source code generation, so it doesn't help much.
What I am looking for are the real world usage examples.

**Comment by anton-matosov** [comment]
@LorenzMeier I have implemented the conversion from Quaternion to Euler, but I have not tested this code as not quite sure how to do this.
Can you either help me out or test it yourself, please?

**Comment by LorenzMeier** [comment]
Looks good, there is little chance its not correct. I will try it in the next days, just received my servo gimbal.

**Comment by powerWoW** [comment]
Hey Anton,

I'm testing this ServoGimbal implementation but I'm having some problems: https://groups.google.com/forum/#!topic/px4users/24id7IhwcHM

Could you help me with this?

**Pull request merged**